dra-evolution/testdata/pod-one-container-shared-split-allocation-gpus.yaml

```diff
@@ -1,20 +1,21 @@
 # A Pod asking for 1-8 whole OR partial Foozer GPUs such that the total
 # available memory is at least 128Gi.
 
-apiVersion: devmgmtproto.k8s.io/v1alpha1
-kind: DeviceClaim
+apiVersion: resource.k8s.io/v1alpha2
+kind: ResourceClaimTemplate
 metadata:
-  name: template-foozer-split-shared-128Gi
+  name: foozer-split-shared-128Gi
   namespace: default
 spec:
-  devices:
-  - name: gpu
-    allOf:
-    - class: example.com-foozer
-      maxDevices: 8
-      allowShared: true
-      requests:
-        memory: 128Gi
+  spec:
+    requests:
+    - resourceClassName: example.com-foozer
+      count:
+        maximum: 8
+      requirements:
+      - resource:
+          name: memory
```

Comment:
This is where @klueska and I didn't agree. My immediate reaction when seeing this was "who defines what `memory` means"?

@klueska explained that the expectation is that the class or device filter narrow down the instance candidates down to devices from one vendor, so that vendor defines `memory` and the vendor driver name suffix is not needed.

The risk then is that if a) the filter allows devices from different vendors and b) different vendors define `memory` differently, the resource requirement might be satisfied in unexpected ways.

I prefer to require that resource names are fully-qualified the same way as attributes.

Reply:
But as I said in https://github.com/kubernetes-sigs/wg-device-management/pull/14/files#r1605246207, this is not a block for merging this prototype.

Reply:
> All attributes are fully qualified with a suffix that is either (1) the driver name or (2) a standardized suffix, e.g. dra.k8s.io. Examples include pcie-root.dra.k8s.io or product-name.gpu.nvidia.com.

For easy of use in the CEL expressions, I would probably prefer standard ones to have no suffix, for readability. And for those to follow json naming conventions so that we can do "device.pcieRoot" instead of "device.attributes['pcie-root.dra.k8s.io']".

> In addition to attributes, devices in a ResourcePool can include a set of resources (I believe you called these sharedResourcesConsumed in the k8srm-prototype). These can either be local to the device that is defining them, or pulled from a shared pool (i.e. the sharedResources construct from the k8srm-prototype). It is these constructs that are being discussed here, not attributes.

I still don't understand what "local to the device" is here. Is that how you are referring to what are called "claimResourcesProvided" in the other prototype? I think mixing these things in this "source" object conflates two things that are really quite different.

I definitely agree with Kevin on the resource names for drawing down on the shared resource pool in the device partitioning case. Those only need to be internally consistent within that specific device, so no qualification is needed.

Those are distinct (to me) from resources provided for the claim. There I think we may end up needing qualified resource names there, since the semantics are possibly different. But I am not 100% sure either way yet.